pyxem/signals/beam_shift.py

```diff
@@ -43,7 +43,13 @@ def make_linear_plane(self, **kwargs):
         self.data = s_linear_plane.data
         self.events.data_changed.trigger(None)
 
-    def get_linear_plane(self, mask=None, fit_corners=None):
+    def get_linear_plane(
+        self,
+        mask=None,
+        fit_corners=None,
+        initial_values=None,
+        constrain_magnitude=False,
```

Comment:
`constrain_magnitude` is a bit imprecise, as I reread it. To be precise, it actually minimises the magnitude variance. So I think I will rename it `constrain_magnitude_variance` to be more descriptive.

And I think it will not work well by default as it does not pay attention to the actual value of the magnitude, only its variance. When fitting a linear plane to a non-electromagnetic area, which is the most common use case, you actually want to minimise the magnitude itself, which it would not try to do if `constrain_magnitude_variance` is `True`. Maybe there would be a good way to minimise both the magnitude and variance at the same time, with a better cost function or some such, but it does not seem obvious for me.